src/components/proto/menu.svelte

```diff
@@ -1,128 +1,200 @@
 <script lang="ts">
-  import { onMount } from 'svelte';
-
-  onMount(() => {
-    const openButton = document.getElementById('open-button');
-    const backButton = document.getElementById('back-button');
-    const menu = document.getElementById('menu');
-    if (openButton != null) {
-      openButton.onclick = () => {
-        if (menu != null) menu.style.display = '';
-      };
+  let menuOpen = false;
+  let menuStack = ['menu-contents-0'];
+  let menuStackLength = menuStack.length;
+  let visibleClass = 'visible';
+
+  const openMenu = () => {
+    menuOpen = true;
+  };
+
+  const closeMenu = () => {
+    menuOpen = false;
+  };
+
+  const push = (id: string) => {
```

Comment:
Very good contender for unit testable functions: `push` and `pop`

Reply:
I'll make these integrated tests instead since there's lots of dependencies on the dom. I'll do this once I handle passing menus as props.